### ISSUE TITLE
What would be the proper way to set support of net & gross prices?

### ISSUE THREAD

**Issue by NyanKiyoshi** [mirumee/saleor#1751]
I'm currently trying to see for a way to integrate net & gross support. But since I'm not really familiar for the moment with saleor's code, I wanted to hear your thoughts about this.

For now, in my own case, it doesn't matter to really take into account the gross (for example, in Europe, since a few years, as you may probably know, an European buyer must (or should? I don't know how they are enforcing that) pay the tax of its country, instead of paying the enterprise/org/...'s country tax rate).

So, since right now the project is mostly national and with another close country (1% difference between the two), it may be okay to use only our country's tax rate (so by using the gross price field).

Then, the simpler solution and the cleanest (because of diffs and merging, to avoid having too many conflicts) since saleor is moving really fast, would be to use simple tags or filters to get both the gross and net price. So, no real logic behind, just for the eyes of the customers.

My other solution (which is the one currently used), is to inherit `prices.Price` to integrate a tax calculation which is more extensible and is not making too much diff or issues of compatibility (clean code FTW).

I'm really hoping to get your thoughts about this, I'm posting this here, hoping maybe to have the feature directly in the core of saleor?

Thanks again for your hard work!

Edit: I'm also thinking that I will also need to store the net and gross total of the order for the invoice as well. I'll look all this tomorrow for how saleor works on that and the prices library that is currently used.

**Comment by rafalp** [comment]
Hiya Nyan!

You are correct that European buyers have to pay value added tax AKA VAT, with different goods having different vat rates between eu countries with businesses being extempt from it. Being EU based ourselves we understand importance of getting the proper tax handling in saleor's core, and while groundwork is already happening on this in form of updated `prices` and `django-prices` that we still haven't updated the Saleor to work with, AFAIK we still haven't settled on how taxes should ultimately be implemented.

Making `prices.Price` tax aware is actually the course we've set with latest changes in `prices` where API has been changed to implement `TaxedMoney` having `net` and `gross` attributes. Going from there we had rough idea that products prices entered in dashboard could be either all gross or all net depending on how your store is setup and fits your business, with tax computation going from there.

As for going with vat computation for just one country because there's 1% difference, this feels... *dangerous* and its hard for us to recommend with clear consious consciousness. ;) On checkout user picks the shipping country, so its good time to compute the valid gross/net here.

> I'm also thinking that I will also need to store the net and gross total of the order for the invoice as well.

Yup, EU requires invoices to contain information required to calculate amount or rate of tax used, so its good to keep those for historic reasons. We are already storing product's net and gross prices for each individual order for invoices show prices from time of order instead of current ones, so mechanic to expand upon is here for you.
